File: dsl-style-java/src/main/java/example/dsl_style_java/DbConfig.java
package example.dsl_style_java;

import java.util.Objects;
import javax.sql.DataSource;
import org.seasar.doma.jdbc.Config;
import org.seasar.doma.jdbc.JdbcLogger;
import org.seasar.doma.jdbc.dialect.Dialect;
import org.seasar.doma.jdbc.tx.TransactionManager;

public class DbConfig implements Config {

  private final Dialect dialect;
  private final DataSource dataSource;
  private final JdbcLogger jdbcLogger;
  private final TransactionManager transactionManager;

  public DbConfig(
      Dialect dialect,
      DataSource dataSource,
      JdbcLogger jdbcLogger,
      TransactionManager transactionManager) {
    this.dialect = Objects.requireNonNull(dialect);
    this.dataSource = Objects.requireNonNull(dataSource);
    this.jdbcLogger = Objects.requireNonNull(jdbcLogger);
    this.transactionManager = Objects.requireNonNull(transactionManager);
  }

  @Override
  public JdbcLogger getJdbcLogger() {
    return jdbcLogger;
  }

  @Override
  public Dialect getDialect() {
    return dialect;
  }

  @Override
  public DataSource getDataSource() {
    return dataSource;
  }

  @Override
  public TransactionManager getTransactionManager() {
    return transactionManager;
  }
}

File: dsl-style-kotlin/src/main/kotlin/example/dsl_style_kotlin/DbConfig.kt
package example.dsl_style_kotlin

import org.seasar.doma.jdbc.Config
import org.seasar.doma.jdbc.JdbcLogger
import org.seasar.doma.jdbc.dialect.Dialect
import org.seasar.doma.jdbc.tx.TransactionManager
import javax.sql.DataSource

class DbConfig(
    private val dialect: Dialect,
    private val dataSource: DataSource,
    private val jdbcLogger: JdbcLogger,
    private val transactionManager: TransactionManager
) : Config {
    override fun getJdbcLogger(): JdbcLogger {
        return jdbcLogger
    }

    override fun getDialect(): Dialect {
        return dialect
    }

    override fun getDataSource(): DataSource {
        return dataSource
    }

    override fun getTransactionManager(): TransactionManager {
        return transactionManager
    }
}
